### PR TITLE
More localized error handling

### PR DESCRIPTION
**Pull request by zah**

### CODE REVIEW

File: metrics/chronos_httpserver.nim [hunk 1]
```diff
@@ -65,7 +65,10 @@ when defined(metrics):
           printError(exc.msg)
 
     let socketFlags = {ServerFlags.ReuseAddr}
-    let res = HttpServerRef.new(address, cb, socketFlags = socketFlags)
+    let res = try: HttpServerRef.new(address, cb, socketFlags = socketFlags)
```
Comment (cheatfate) [inline metrics/chronos_httpserver.nim:68]: `new` procedure do not produce `CatchableError` anymore, i have workaround Nim's issue in https://github.com/status-im/nim-chronos/commit/f2e4d447d6aec99b3641d51994650769c5c00d02

File: metrics/chronos_httpserver.nim [hunk 2]
```diff
@@ -305,49 +305,54 @@ proc new*(t: typedesc[MetricsHttpServerRef], address: string,
...
           return err("Invalid server address")
+    var
       request =
         block:
           let res = createAsyncPipe()
           if (res.read == asyncInvalidPipe) or (res.write == asyncInvalidPipe):
             return err("Unable to create communication request pipe")
           res
+      cleanupRequest = true
+    defer:
```
Comment (arnetheduck) [inline metrics/chronos_httpserver.nim:316]: I'm not convinced this defer is a win when reading the code - ie the booleands do quite a good job of obfuscating the control flow, which also can be seen in the line count

Comment (zah) [reply metrics/chronos_httpserver.nim:316]: I strongly disagree. With this structure in place it's trivial to modify the control flow in the function in safe manner. Without it, one has to be extremely careful to keep all the exit paths correct.

Comment (zah) [reply metrics/chronos_httpserver.nim:316]: You should appreciate this classic article and the arguments that it puts forward:
http://ptgmedia.pearsoncmg.com/imprint_downloads/informit/aw/meyerscddemo/DEMO/MAGAZINE/CA_FRAME.HTM

Resource management without RAII-like patterns in just not viable.

**Pull request merged**